internals/overlord/cmdstate/request.go

```diff
@@ -70,8 +70,14 @@ func Exec(st *state.State, args *ExecArgs) (*state.Task, ExecMetadata, error) {
 		return nil, ExecMetadata{}, errors.New("cannot use interactive mode without a terminal")
 	}
 
-	// Inherit the pebble daemon environment.
+	// Inherit the pebble daemon environment (except HOME and USER env vars)
```

Comment:
I think it's worth expanding this comment, explaining why it was necessary. Also, if `HOME` and `USER` aren't present in `args.Environment`, they are set up below based on the username and home directory of the provided user ID (or the current user if not specified).

Reply:
addressed in [88a02c5](https://github.com/canonical/pebble/pull/262/commits/88a02c5938164e08d2f8380297ce24727c695a49)